aptos-move/framework/src/natives/aggregator_natives/context.rs

```diff
@@ -164,31 +174,73 @@ mod test {
     // All aggregators are initialized deterministically based on their ID,
     // with the following spec.
     //
-    //     +-------+---------------+----------+-----+---------+
-    //     |  key  | storage value |  create  | get | remove  |
-    //     +-------+---------------+----------+-----+---------+
-    //     |  100  |               |   yes    | yes |   yes   |
-    //     |  200  |               |   yes    | yes |         |
-    //     |  300  |               |   yes    |     |   yes   |
-    //     |  400  |               |   yes    |     |         |
-    //     |  500  |               |          | yes |   yes   |
-    //     |  600  |               |          | yes |         |
-    //     |  700  |               |          | yes |         |
-    //     |  800  |               |          |     |   yes   |
-    //     +-------+---------------+----------+-----+---------+
-    fn test_set_up(context: &NativeAggregatorContext) {
+    //     +-------+---------------+-----------------------------------+-----+---------+
+    //     |  key  | storage value |  create                           | get | remove  |
+    //     +-------+---------------+-----------------------------------+-----+---------+
+    //     |  100  |               |   yes                              | yes |   yes   |
+    //     |  200  |               |   yes                              | yes |         |
+    //     |  300  |               |   yes                              |     |   yes   |
+    //     |  400  |               |   yes                              |     |         |
+    //     |  500  |               |   Yes if !aggregagtor_enabled      | yes |   yes   |
+    //     |  600  |               |   Yes if !aggregagtor_enabled      | yes |         |
+    //     |  700  |               |   Yes if !aggregagtor_enabled      | yes |         |
+    //     |  800  |               |                                    |     |   yes   |
+    //     +-------+---------------+------------------------------------+-----+---------+
+    fn test_set_up(context: &NativeAggregatorContext, aggregator_enabled: bool) {
         let mut aggregator_data = context.aggregator_data.borrow_mut();
 
         aggregator_data.create_new_aggregator(aggregator_id_for_test(100), 100);
         aggregator_data.create_new_aggregator(aggregator_id_for_test(200), 200);
         aggregator_data.create_new_aggregator(aggregator_id_for_test(300), 300);
         aggregator_data.create_new_aggregator(aggregator_id_for_test(400), 400);
+        if !aggregator_enabled {
```

Comment:
This is wrong: instead of creating an aggregator in storage (adding some data to `EmptyStorage`) you make creation of aggregators optional. So test you mentioned above fails for a good reason. Also this never goes to the path in `read_and_materialize` where storage has something.

Reply:
Done.